haystack/nodes/retriever/dense.py

```diff
@@ -42,7 +43,42 @@
 logger = logging.getLogger(__name__)
 
 
-class DensePassageRetriever(BaseRetriever):
+class DenseRetriever(BaseRetriever):
+    """
+    Base class for all dense retrievers.
+    """
+
+    @abstractmethod
+    def embed_queries(self, queries: List[str]) -> np.ndarray:
+        """
+        Create embeddings for a list of queries.
+
+        :param queries: List of queries to embed.
+        :return: Embeddings, one per input query, shape: (queries, embedding_dim)
+        """
+        pass
+
+    @abstractmethod
+    def embed_documents(self, documents: List[Document]) -> np.ndarray:
+        """
+        Create embeddings for a list of documents.
+
+        :param documents: List of documents to embed.
+        :return: Embeddings of documents, one per input document, shape: (documents, embedding_dim)
+        """
+        pass
+
+    def run_indexing(self, documents: List[Union[dict, Document]]):
+        documents = deepcopy(documents)
+        document_objects = [Document.from_dict(doc) if isinstance(doc, dict) else doc for doc in documents]
+        embeddings = self.embed_documents(document_objects)
+        for doc, emb in zip(document_objects, embeddings):
+            doc.embedding = emb
```

Comment:
Ah no :smile: I think that the first two lines can be safely removed. (see https://github.com/deepset-ai/haystack/pull/3243)
```suggestion
        embeddings = self.embed_documents(documents)
        for doc, emb in zip(document_objects, embeddings):
            doc.embedding = emb
```